### PR TITLE
Lookup SMS numbers before sending

### LINKED ISSUE #1082: twilio lookup
### TL;DR

Investigate using the lookup API: https://www.twilio.com/docs/lookup
See if this would reduce our false positive SMS sends

/kind enhancement

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes #1082

**Release Note**

```release-note
Ensure phone numbers are capable of receiving SMS messages before sending
```

/assign @mikehelmick

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -176,7 +176,9 @@ type IssueCodeRequest struct {
 	// Offset in minutes of the user's timezone. Positive, negative, 0, or omitted
 	// (using the default of 0) are all valid. 0 is considered to be UTC.
 	TZOffset float32 `json:"tzOffset"`
-	Phone    string  `json:"phone"`
+
+	// Phone is the E.164 formatted phone number.
```
Comment (mikehelmick) [inline pkg/api/api.go:180]: can you add this comment to the API docs as well?

Comment (sethvargo) [reply pkg/api/api.go:180]: Done.

File: pkg/config/admin_server_config.go [hunk 2]
```diff
@@ -50,6 +50,10 @@ type AdminAPIServerConfig struct {
 	AllowedSymptomAge   time.Duration `env:"ALLOWED_PAST_SYMPTOM_DAYS,default=660h"` // 660h is 28 days.
 	EnforceRealmQuotas  bool          `env:"ENFORCE_REALM_QUOTAS, default=true"`
 
+	// ValidateSMSNumbers instructs the server to validate phone numbers are
+	// capable of receiving SMS messages before attempting delivery.
+	ValidateSMSNumbers bool `env:"VALIDATE_SMS_NUMBERS, default=true"`
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply pkg/config/admin_server_config.go]: Done.

**Comment by sethvargo** [comment]
@mikehelmick made it a realm-level configuration

**Comment by sethvargo** [comment]
We've found some situations in which this provides false negatives, which isn't feasible given our use.

**Pull request closed without merging**